Library/Extensions/NfieldConnectionClientExtensions.cs

```diff
@@ -34,39 +33,36 @@ private static Uri BackgroundActivityUrl(this INfieldConnectionClient client, st
         }
 
         /// <summary>
-        /// Recursive method that polls the activity status until it completes.
+        /// Method that polls the activity status until it completes.
         /// </summary>
         /// <param name="connectionClient">NField Connection Client (this)</param>
         /// <param name="activityId">The id of the activity to wait for.</param>
         /// <param name="fieldNameResult">The name of the result field</param>
         /// <returns>The <see cref="BackgroundActivityStatus" /> id.</returns>
-        internal static Task<T> GetActivityResultAsync<T>(this INfieldConnectionClient connectionClient, string activityId, string fieldNameResult)
+        internal static async Task<T> GetActivityResultAsync<T>(this INfieldConnectionClient connectionClient, string activityId, string fieldNameResult)
         {
-            return connectionClient.Client.GetAsync(connectionClient.BackgroundActivityUrl(activityId))
-                .ContinueWith(response => response.Result.Content.ReadAsStringAsync())
-                .Unwrap()
-                .ContinueWith(content =>
-                {
-                    var obj = JObject.Parse(content.Result);
-                    var status = obj["Status"].Value<int>();
+            while (true)
+            {
+                var response = await connectionClient.Client.GetAsync(connectionClient.BackgroundActivityUrl(activityId));
```

Comment:
shouldn't we be adding `.ConfigureAwait(false)` to all async calls?

Reply:
added